src/main/java/com/google/devtools/build/lib/skyframe/ConfiguredTargetFunction.java

```diff
@@ -440,6 +542,62 @@ public SkyValue compute(SkyKey key, Environment env)
     }
   }
 
+  // Taken from ConfiguredTargetFactory.convertVisibility().
+  // TODO(phil): Figure out how to de-duplicate somehow.
```

Comment:
@gregestren , can you help me figure out how to de-duplicate this function?

I can _almost_ do it. The part I'm struggling with is figuring out how to deal with the `PackageGroupsRuleVisibility` section. The thing that stumps me is this line here: https://github.com/bazelbuild/bazel/blob/7f74878b9da57666f04cb607a3d7a36122143863/src/main/java/com/google/devtools/build/lib/analysis/ConfiguredTargetFactory.java#L158

What's the significance of the configuration being `null`? I wanted to query the `visibility` labels similar to how I am currently querying the `target_compatible_with` labels above. That, however, I'm doing with this: https://github.com/bazelbuild/bazel/pull/14096/files#diff-6a800890b53e39d722c5457065ec7694a044c12b6f8ac93b2147c15ab7ac95f8R362-R366
That has a non-null configuration.

I'll dig into it more. I'm probably overlooking something simple.

Reply:
Noted. Let's hold off on this pending my file comments.